plugins/InspectionGadgets/InspectionGadgetsAnalysis/src/com/siyeh/ig/bugs/IncorrectDateTimeFormatInspection.java

```diff
@@ -0,0 +1,233 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.siyeh.ig.bugs;
+
+import com.intellij.codeInspection.AbstractBaseJavaLocalInspectionTool;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.*;
+import com.siyeh.InspectionGadgetsBundle;
+import com.siyeh.ig.callMatcher.CallMatcher;
+import com.siyeh.ig.psiutils.ExpressionUtils;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.IntPredicate;
+
+import static com.siyeh.ig.callMatcher.CallMatcher.*;
+
+public class IncorrectDateTimeFormatInspection extends AbstractBaseJavaLocalInspectionTool {
+
+  private static IntPredicate rangeOf(int from, int to) {
+    return count -> count >= from && count <= to;
+  }
+
+  private static IntPredicate setOf(int... numbers) {
+    return  count -> IntOpenHashSet.of(numbers).contains(count);
```

Comment:
This way you recreate the set every time lambda is invoked, which looks excessive. I suggested to use method reference to create it once. Alternatively, you can get rid of the set at all and use linear search (taking into account that your sets are really small):
`count -> ArrayUtil.indexOf(numbers, count) >= 0`

Reply:
Shame on me. Thank you! `ArrayUtil.indexOf(numbers, count) >= 0` is much faster than Set.  Fixed